### ISSUE TITLE
Potentially expensive indexes built on every feed

### ISSUE THREAD

**Issue by abyrd** [conveyal/gtfs-lib#29]
I see messages indicating that various indexes (e.g. an index from stops to stoptimes) are being built every time a feed is opened. This might make sense for use in our GTFS cache and data tools projects, but we should keep in mind that the basic use of gtfs-lib is just for building graphs in R5. These indexes should be lazy-initialized or initialized only on request to avoid slow startup times and high memory consumption in applications which don't need them.

**Comment by abyrd** [comment]
@landonreed I think maybe you created these indexes. This is one of the big ones:

```
LOG.info("Building stop to stop times index");
        Bind.histogram(stop_times, stopCountByStopTime, (key, stopTime) -> stopTime.stop_id);
        Bind.secondaryKeys(stop_times, stopStopTimeSet, (key, stopTime) -> new String[] {stopTime.stop_id});
```
It takes a lot of time and memory to create in a big network like the Netherlands, which means the wait to being routing or analysis is longer than it would need to be. We should pull the creation of these indexes out into functions that can be called optionally, only in applications that need the indexes.

Then again, if we move all this data into a more standard DBMS environment we may be able to keep the R5 backend really simple and not offer these indexes at all.

Also, the field names / table names are not very easy to interpret.

**Comment by landonreed** [comment]
@abyrd, yes we created those for some of the statistics that are being calculated in datatools and gtfs-api.  We can certainly move them to be lazy-initialized if that doesn't break the mapDB files and yes, I'm also hoping that moving to a new database system will make indexes like these easier to manage.

Did you need an immediate action on this issue?

**Comment by abyrd** [comment]
No, not immediate, just wanted your opinion on the subject and wanted to register it for future action. Thanks @landonreed .

**Comment by abyrd** [comment]
@ansoncfit fixed this on a maintenance branch with ea0776c3056353ff58741a472be51ab991139aae. That patch which is in release 3.0.1 still needs to make it into master.
